src/libosd/terminal.c

```diff
@@ -0,0 +1,281 @@
+/* Copyright 2018 The Open SoC Debug Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <osd/cl_dem_uart.h>
+#include <osd/terminal.h>
+
+#include <assert.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <pthread.h>
+#include <osd/osd.h>
+#include "osd-private.h"
+
+/**3 possible ways to connect to the the DEM-Uart module
+ * 1. Use the current stdin/stdout for this process
+ * 2. Open a new terminal window (using system("konsole") or openypty() or posix_openpt()) and write to that.
+ * 	  BSD or System V ptys?
+ * 3. Create a new device file that can be accessed by tools such as minicom?
```

Comment:
Option 3 is the one we're looking for, as discussed in ticket #24. However, for now just printing the received characters is a good start.